integration-tests/tests/utils/docker_utils.py

```diff
@@ -0,0 +1,72 @@
+"""Provide utility functions related to the use of Docker during integration tests."""
+
+import re
+import shutil
+import subprocess
+from enum import Enum
+
+
+class DockerStatus(str, Enum):
+    """Enum of possible Docker Statuses."""
+
+    created = "created"
+    restarting = "restarting"
+    running = "running"
+    removing = "removing"
+    paused = "paused"
+    exited = "exited"
+    dead = "dead"
+
+
+def get_docker_binary_path() -> str:
+    """
+    Finds the absolute path to the Docker client in the current PATH.
+
+    :return: Absolute path to the Docker binary.
+    :raise RuntimeError: docker is not found on PATH.
+    """
+    docker_bin = shutil.which("docker")
+    if docker_bin is None:
+        err_msg = "docker not found in PATH"
+        raise RuntimeError(err_msg)
+    return docker_bin
```

Comment:
this is great stuff. shall we move it into clp-py-utils so anywhere that creates docker subprocess can use this helper?

Reply:
This function could be generalized to get the path for any binary that exists on `$PATH`. This function is mainly  used to address ruff S607: https://docs.astral.sh/ruff/rules/start-process-with-partial-path/